File: app/src/main/java/ru/melodin/fast/common/AppGlobal.java
package ru.melodin.fast.common;

import android.app.Application;
import android.content.SharedPreferences;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.database.sqlite.SQLiteDatabase;
import android.os.Handler;
import android.preference.PreferenceManager;

import java.util.Locale;

import ru.melodin.fast.database.DatabaseHelper;

public class AppGlobal extends Application {

    private static AppGlobal instance;

    public static volatile SQLiteDatabase database;
    public static volatile Locale locale;
    public static volatile Handler handler;
    public static volatile SharedPreferences preferences;

    public static volatile String app_version_name;
    public static volatile int app_version_code;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        handler = new Handler(getMainLooper());
        preferences = PreferenceManager.getDefaultSharedPreferences(this);
        database = DatabaseHelper.getInstance().getWritableDatabase();
        locale = Locale.getDefault();

        try {
            PackageInfo pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
            app_version_name = pInfo.versionName;
            app_version_code = pInfo.versionCode;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

        CrashManager.init();
        ThemeManager.init();
    }

    public static synchronized AppGlobal context() {
        return instance;
    }
}

File: app/src/main/java/ru/melodin/fast/common/AppGlobal.kt
package ru.melodin.fast.common

import android.app.Application
import android.content.ClipboardManager
import android.content.Context
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.content.res.Resources
import android.database.sqlite.SQLiteDatabase
import android.net.ConnectivityManager
import android.os.Handler
import android.view.inputmethod.InputMethodManager
import androidx.core.content.pm.PackageInfoCompat
import androidx.preference.PreferenceManager
import ru.melodin.fast.database.DatabaseHelper
import java.util.*

class AppGlobal : Application() {

    companion object {

        lateinit var database: SQLiteDatabase

        lateinit var locale: Locale

        lateinit var handler: Handler

        lateinit var preferences: SharedPreferences

        var app_version_name = ""

        var app_version_code = -1

        lateinit var clipService: ClipboardManager

        lateinit var connectionService: ConnectivityManager

        lateinit var inputService: InputMethodManager

        lateinit var res: Resources

        fun isAlpha(): Boolean {
            return app_version_name.toLowerCase(Locale.getDefault()).contains("alpha")
        }

        fun isBeta(): Boolean {
            return app_version_name.toLowerCase(Locale.getDefault()).contains("beta")
        }

        fun isDebug(): Boolean {
            return isAlpha() || isBeta()
        }
    }

    override fun onCreate() {
        super.onCreate()
        res = resources
        handler = Handler(mainLooper)
        preferences = PreferenceManager.getDefaultSharedPreferences(this)
        database = DatabaseHelper.getInstance(this).writableDatabase
        locale = Locale.getDefault()

        try {
            val pInfo = packageManager.getPackageInfo(packageName, 0)
            app_version_name = pInfo.versionName
            app_version_code = PackageInfoCompat.getLongVersionCode(pInfo).toInt()
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }

        initServices()
        CrashManager.init()
        TaskManager.init()
        ThemeManager.init()
    }

    private fun initServices() {
        inputService = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        connectionService = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        clipService = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    }


}
